conda_index/index/fs.py

```diff
@@ -0,0 +1,90 @@
+"""
+Minimal (just what conda-index uses) filesystem abstraction.
+
+Allows `fsspec <https://filesystem-spec.readthedocs.io/>`_ to be used to index
+remote repositories, without making it a required dependency.
+"""
+
+from __future__ import annotations
+
+import os
+import os.path
+import typing
+from dataclasses import dataclass
+from numbers import Number
+from pathlib import Path
+
+if typing.TYPE_CHECKING:  # pragma: no cover
+    from fsspec import AbstractFileSystem
+
+
+# Note fsspec uses / as a path separator on all platforms
+
+
+@dataclass
+class FileInfo:
+    """
+    Filename and a bit of stat information.
+    """
+
+    fn: str
+    st_mtime: Number
+    st_size: Number
+
+
+class MinimalFS:
+    """
+    Filesystem API as needed by conda-index, for fsspec compatibility.
+    """
+
+    def open(self, path: str, mode: str = "rb"):
+        return Path(path).open(mode)
+
+    def stat(self, path: str):
+        st_result = os.stat(path)
```

Comment:
For almost all of the calls to `MinimalFS.open` (and `MinimalFS.stat`) do something like:

```python
abs_fn = self.fs.join(...)
self.fs.open(abs_fn)  # or self.fs.stat(abs_fn)
```

Thoughts on combining and doing the join here?

```suggestion
    def open(self, *path: str, mode: str = "rb"):
        return Path(*path).open(mode)

    def stat(self, *path: str):
        st_result = Path(*path).stat()
```

Reply:
Would rather keep it more similar to `os.path`